gcsa/recurrence.py

```diff
@@ -474,13 +483,20 @@ def to_string(l):
         if freq not in (HOURLY, MINUTELY, DAILY, WEEKLY, MONTHLY, YEARLY):
             raise ValueError('"freq" parameter must be one of HOURLY, MINUTELY, DAILY, WEEKLY, MONTHLY or YEARLY. '
                              '{} was provided'.format(freq))
-        if interval and (isinstance(interval, int) or interval < 1):
+        if interval and (not isinstance(interval, int) or interval < 1):
             raise ValueError('"interval" parameter must be a positive int. '
                              '{} was provided'.format(interval))
         if count and (not isinstance(count, int) or count < 1):
             raise ValueError('"count" parameter must be a positive int. '
                              '{} was provided'.format(count))
 
+        if until:
+            if not isinstance(until, (date, datetime)):
+                msg = 'The until object must be a date or datetime, not {!r}.'.format(end.__class__.__name__)
```

Comment:
Should be `until` instead of `end`.

Reply:
Good catch! Copy-n-paste weakness...
It also was my first pull request on github. ;)
Thanks!